internal/landing/landing.go

```diff
@@ -8,22 +8,74 @@ package landing
 
 import (
 	"embed"
+	"fmt"
+	"html/template"
 	"io/fs"
 	"net/http"
+	"path/filepath"
+
+	"github.com/talos-systems/discovery-service/internal/state"
 )
 
 //go:embed "html/index.html"
 var static embed.FS
 
+type ClusterData struct {
+	ClusterId  string
+	Affiliates []*state.AffiliateExport
+}
+
+func InspectHandler(w http.ResponseWriter, req *http.Request, state *state.State) {
+
+	clusterData := ClusterData{}
+
+	if req.Method == "POST" {
+		if err := req.ParseForm(); err != nil {
+			panic(err)
+		}
+		if formData := req.PostForm["cluster-id"]; formData != nil {
+			clusterData.ClusterId = formData[0]
+		}
+		fetchedCluster := state.GetCluster(clusterData.ClusterId)
+
+		clusterData.Affiliates = fetchedCluster.List()
+	}
+
+	file, err := filepath.Abs("internal/landing/html/inspect.html")
```

Comment:
this won't work:

1. No need to do `filepath.Abs` anyways - relative path would have worked if it was there
2. Source code is not available to the compiled binary.
3. As you can see above, files are embedded into the Go binary to the variable `static` of type `embed.FS`. 
4. It is used on its own to display static files, but you can pull back any file with `static.ReadFile`.
5. In this particular case I think it's easier to keep the template separate (in other file outside of `html`):

```go
//go:embed templates/inspect.html.tmpl
var inspectTemplate []byte
```

see https://pkg.go.dev/embed

Reply:
this way `inspectTemplate` variable will contain template contents in runtime.